.github/workflows/release_parallel.yml

```diff
@@ -0,0 +1,218 @@
+                  npm install
+                  rm -rf node_modules
+
+            # TODO we should use some Action-specific bot account
+            - name: Commit and push changes
+              run: |
+                  git config --global user.name 'Jordan Last'
+                  git config --global user.email 'jordan.michael.last@gmail.com'
+                  git config --global commit.gpgsign true
+                  echo -n "$GPG_SIGNING_KEY" | base64 --decode | gpg --import
+                  git config --global user.signingkey C8B77BCBE16CD2B94B43F9C8757397B82D4ED7B0
+                  BRANCH_NAME="update-${{ needs.prepare-release.outputs.release-version }}-$(echo '${{ matrix.test.displayPath }}' | sed 's/\//-/g')"
+                  git switch -c "$BRANCH_NAME"
+                  git add --all
+                  if ! git diff --cached --quiet; then
+                    git commit -m "Update dependencies for ${{ matrix.test.displayPath }}"
+                  else
+                    echo "No changes to commit. Skipping commit and push."
+                  fi
+                  git push origin "$BRANCH_NAME"
+
+    finalize-release:
+        needs: [prepare-release, update-dependencies]
+        name: Finalize Release
+        runs-on: ubuntu-latest
+        env:
+            GPG_SIGNING_KEY: ${{ secrets.GPG_SIGNING_KEY }}
+            GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        steps:
+            - uses: actions/checkout@v4
+              with:
+                  ref: ${{ github.event.pull_request.head.ref || github.ref }}
+                  token: ${{ secrets.LASTMJS_GITHUB_TOKEN }}
+                  fetch-depth: 0 # Fetch all history for all branches and tags
+
+            - name: Merge update branches
+              env:
+                  PAT: ${{ secrets.LASTMJS_GITHUB_TOKEN }}
+              run: |
+                  git config --global user.name 'Jordan Last'
+                  git config --global user.email 'jordan.michael.last@gmail.com'
+                  git config --global commit.gpgsign true
+                  echo -n "$GPG_SIGNING_KEY" | base64 --decode | gpg --import
+                  git config --global user.signingkey C8B77BCBE16CD2B94B43F9C8757397B82D4ED7B0
+
+                  # Determine the current branch name
+                  CURRENT_BRANCH=$(git rev-parse --abbrev-ref HEAD)
+                  echo "Current branch: $CURRENT_BRANCH"
+
+                  # Fetch all branches
+                  git fetch --all
+
+                  # List and merge update branches
+                  for branch in $(git branch -r | grep "origin/update-${{ needs.prepare-release.outputs.release-version }}-" | sed 's/origin\///'); do
+                    echo "Merging branch: $branch"
+                    git fetch origin $branch:$branch
+                    git merge --no-ff "$branch" -m "Merge $branch" || {
```

Comment:
Are we sure a merge is the best strategy here?

Reply:
I was until you asked me about it

Reply:
Could we explore squashing all of them?

Reply:
could a rebase even without a squash result in half the commits because you won't have each commit followed by a merge commit?

Reply:
done